lib/company/opening.js

```diff
@@ -348,10 +348,231 @@ function getAllOpeningsOfCompany(userId, callback) {
+      },
+    },
+    function(err, updationResponse) {
+      if (err) {
+        callback({
+          type: CONSTANTS.ERROR_TYPES.DB_ERROR,
+          msg: 'Failed to fetch Interview Track Details',
+          errorDetail: JSON.stringify(err),
+        });
+      } else if (selected.length != updationResponse.nModified) {
+        callback({
+          type: CONSTANTS.ERROR_TYPES.INCORRECT_PAYLOAD,
+          msg: 'Failed to modify interview status of some record',
+          errorDetail: 'Some record was left unupdated in database',
+        });
+      } else {
+        callback(null, updationResponse);
+      }
+    }
+  );
+}
+
+/**
+ * Function to change interview status of candidates who are rejected in interview
+ * @param {Array} selected array of interview tracks _id and email of rejected candidates
+ * @param {Function} callback Function with two parameters - err and updation Response
+ */
+function changeRejectedStatus(rejected, callback) {
+  Models.InterviewTrack.updateMany(
+    { _id: { $in: rejected } },
+    {
+      $set: {
+        interview_status: CONSTANTS.ENUMS.USER.INTERVIEW_STATUS.REJECTED,
+      },
+    },
+    function(err, updationResponse) {
+      if (err) {
+        callback({
+          type: CONSTANTS.ERROR_TYPES.DB_ERROR,
+          msg: 'Failed to fetch Interview Track Details',
+          errorDetail: JSON.stringify(err),
+        });
+      } else if (rejected.length != updationResponse.nModified) {
+        callback({
+          type: CONSTANTS.ERROR_TYPES.INCORRECT_PAYLOAD,
+          msg: 'Failed to modify interview status of some record',
+          errorDetail: 'Some record was left unupdated in database',
+        });
+      } else {
+        callback(null, updationResponse);
+      }
+    }
+  );
+}
+
+/**
+ * Function to change interview status of all candidates applied in opening and send respective mails to them as per their status
+ * @param {ObjectId} userId _id of User
+ * @param {ObjectId} openingId _id of Opening
+ * @param {Integer} count Number of candidates that company wants to hire
+ * @param {*} callback Function with two parameters - err and Data
+ */
+function hiringCount(userId, openingId, count, callback) {
+  async.waterfall(
+    [
+      function(waterfallCallback) {
+        userHelper.getCompanyDetailsFromUser(userId, function(
+          err,
+          fetchedCompany
+        ) {
+          waterfallCallback(err, fetchedCompany);
+        });
+      },
+
+      function(fetchedCompany, waterfallCallback) {
+        getResultDetail(fetchedCompany._id, openingId, function(
+          err,
+          fetchedOpening
+        ) {
+          waterfallCallback(err, fetchedOpening);
+        });
+      },
+
+      function(fetchedOpening, waterfallCallback) {
+        let selected = [];
+        let rejected = [];
+        if (fetchedOpening.interviewees.length > count) {
+          for (let i = 0; i < count; i++) {
```

Comment:
Instead of using for loop use a map function . And separate this logic for getting list of selected and rejected candidates within another function.  and use that function here. Avoid using code within waterfall use only functions here